src/frontend/gatsby-node.js

```diff
@@ -1,4 +1,44 @@
 const path = require(`path`);
+const fs = require('fs');
+const fetch = require('node-fetch');
+
+exports.onPreInit = () => {
+  const clientId = process.env.unsplashClientId || '';
+  const collectionId = process.env.collectionId || '9975402';
+
+  const frontEndPath = path.join('src', 'images', 'backgrounds');
+
+  if (clientId && collectionId) {
+    fetch(
```

Comment:
This is mixing sync and async calling patterns, and for sure we're going to have intermittent build failures.

I finally figured out that we can use an `async` function with `onPreInit`, see the [caller here](https://github.com/gatsbyjs/gatsby/blob/5037e9d427b585158b40b94ea5ce441bc3044928/packages/gatsby/src/bootstrap/index.js#L188) (which is `await`ing the result) and [this existing code](https://github.com/m-allanson/gatsby-plugin-netlify-cache/blob/88d497975a8a1fb42027838c088d090824e3b3a2/src/gatsby-node.js#L31-L57).

That means we can do the following:

- add `async` on line 5
- add `await` on 12
- move from `then` to `await` on 20 and 21
- rework 39 to be `try/catch`
- get rid of all the `Sync` versions of your calls below.  We don't want to block the main thread like this.

If you're having the `too many open files` bug as a result of this change, we'll deal with that separately: for sure this needs to be done all `async`.